tests/api2/test_305_container.py

```diff
@@ -80,36 +79,93 @@ def test_05_pull_container_image(request):
-        print(result)
         if result['repo_tags'] == [f'{docker_image}:{docker_tag}']:
-            image_id = result['id']
+            private_image_id = result['id']
             assert True, result
             break
     else:
         assert False, results
 
 
-def test_07_get_new_image_with_id(request):
-    depends(request, ["pull_image"])
-    results = GET(f'/container/image/id/{image_id}/')
+def test_07_get_private_image_with_id(request):
+    depends(request, ["pull_private_image"])
+    results = GET(f'/container/image/id/{private_image_id}/')
     assert results.status_code == 200, results.text
     assert isinstance(results.json(), dict), results.text
 
 
-def test_08_get_new_image_with_id(request):
-    depends(request, ["pull_image"])
-    results = DELETE(f'/container/image/id/{image_id}/')
+def test_08_change_enable_image_updates_to_True():
+    payload = {
+        'enable_image_updates': True
+    }
+    results = PUT('/container/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['enable_image_updates'] is True, results
+
+
+@pytest.mark.dependency(name='pull_public_image')
+def test_09_pull_a_public_container_image(request):
+    depends(request, ["setup_kubernetes"], scope="session")
+    payload = {
+        "from_image": 'ixsystems/truecommand',
+        "tag": 'latest'
+    }
+    results = POST('/container/image/pull/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 180)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+def test_10_get_new_public_image_id(request):
+    depends(request, ["pull_public_image"])
+    global public_image_id
+    results = GET("/container/image/")
+    for result in results.json():
+        if result['repo_tags'] == ['ixsystems/truecommand:latest']:
+            public_image_id = result['id']
+            assert True, result
+            break
+    else:
+        assert False, results
+
+
+def test_11_get_public_image_with_id(request):
+    depends(request, ["pull_public_image"])
+    results = GET(f'/container/image/id/{public_image_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+
+
+def test_12_delete_private_image_with_id(request):
+    depends(request, ["pull_private_image"])
+    results = DELETE(f'/container/image/id/{private_image_id}/')
+    assert results.status_code == 200, results.text
+    assert results.json() is None, results.text
+
+
+def test_13_verify_the_private_image_id_is_deleted(request):
+    depends(request, ["pull_private_image"])
+    results = GET(f'/container/image/id/{private_image_id}/')
+    assert results.status_code == 404, results.text
+    assert isinstance(results.json(), dict), results.text
+
+
+def test_14_delete_private_image_with_id(request):
```

Comment:
Let' please update the function name to reflect deletion of the public image, same for below

Reply:
Done